### PR TITLE
EQL: Wrap a verification_exception in an index_not_found_exception in case there is no valid index available

### LINKED ISSUE #63529: Index name pattern not present in the error message
During a REST IT test on a secured cluster, in case the index pattern is an inexistent index, the error message itself is stripped of the index being searched for resulting in a message of the form `{"error":{"root_cause":[{"type":"verification_exception","reason":"Found 1 problem\nline -1:-1: Unknown index [*,-*]"}]` for a search like `/inexistent/_eql/search` whereas the expected is to have the index pattern in the error messge itself.

### PR DESCRIPTION
**Pull request by astefan**
In scenarios where Security is enabled in the cluster, a wildcard defined index pattern that gets resolved in a no-index result loses the original pattern and the error message contains a `*, -*` instead. This change wraps the original verification_exception that gets thrown while checking for valid indices in a index_not_found_exception in the REST layer (where also the original index pattern is available). The provided tests show the actual error messages that get thrown in different scenarios (with/without Security, with/without a wildcard index pattern, existent/inexistent index).
Fixes https://github.com/elastic/elasticsearch/issues/63529.

### CODE REVIEW

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/RestEqlSearchAction.java [hunk 1]
```diff
@@ -59,12 +64,33 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
...
+                        channel.sendResponse(new BytesRestResponse(RestStatus.OK, builder));
+                    } catch (Exception e) {
+                        onFailure(e);
+                    }
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    Exception finalException = e;
+                    if (e instanceof IndexNotFoundException) {
```
Comment (costin) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/RestEqlSearchAction.java:82]: Would be good to add a comment on why this is needed.

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/PreAnalyzer.java [hunk 2]
```diff
@@ -17,9 +18,9 @@
 public class PreAnalyzer {
 
     public LogicalPlan preAnalyze(LogicalPlan plan, IndexResolution indices) {
-        // wrap a potential index_not_found_exception with a VerificationException (expected by client)
         if (indices.isValid() == false) {
```
Comment (costin) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/PreAnalyzer.java:21]: A comment is useful to indicate why the Verification exception is wrapped inside an infe

**Review by costin — APPROVED** [review]
LGTM

**Comment by astefan** [comment]
@elasticmachine update branch

**Pull request merged**